### PR TITLE
Fix date formatting on the event issue template

### PR DESCRIPTION
**Pull request by GBKS**
The template has been wrong, leading to events not being included in the calendar. Encountered this with three different people who created events in the past 1-2 weeks. Was wondering why I saw the issues but nothing pop up in my calendar tools. Turns out it was due to this template being wrong.

### CODE REVIEW

**Review by Bosch-0 — APPROVED** [review]

File: .github/ISSUE_TEMPLATE/design_call.md [hunk 1]
```diff
@@ -8,8 +8,8 @@ assignees: ''
 ---
 
 ```meta
-Date: 2022-MM-DD
-Time: 4:00pm
+Date: 2022-DD-MM
```
Comment (sbddesign) [inline .github/ISSUE_TEMPLATE/design_call.md:11]:
```suggestion
Date: 2022-MM-DD
```

**Review by sbddesign — CHANGES_REQUESTED** [review]
Apologies in advance if this looks like I am training for the Tour de France. I just wanted to double-check this, and ended up going deep. 

How certain are you that this is the cause of events not being included in the calendar? I ask because all of these events appear correctly in my calendar:

- [YYYY-MM-DD, 24 Hour](https://github.com/BitcoinDesign/Meta/issues/328)
- [YYYY-DD-MM, 12 Hour](https://github.com/BitcoinDesign/Meta/issues/355)
- [YYYY-DD-MM, 24 Hour](https://github.com/BitcoinDesign/Meta/issues/363)
- YYYY-MM-DD, 12 Hour
  - *All issues using this time format have since been closed, so they will no longer appear on my calendar. But I have never had an issue with these appearing on my calendar. So I can't pull up concrete proof, only my own memory of these working.*

I dug deeper. It looks like the above 3 links are all conformed to UTC timestamps inside the [iCal file](https://github.com/BitcoinDesign/Meta/blob/calendar/events.ical).

Screen Shot 2022-08-12 at 9 11 55 AM

So I dug deeper. It looks like in the [github events calendar repo](https://github.com/GBKS/github-events-calendar), the README says:

> Only issues with the UTCTime (capitalization does not matter) are used.

And if you dig [into index.js](https://github.com/GBKS/github-events-calendar/blob/master/src/index.js#L59) in that repo, it appears that it is using the`UTCTime` from the parsed markdown.

## Summary

- The calendar generator appears to be using the `UTCTime` field, not the 2 fields you have edited in this file.
- The generated calendar file appears to correctly include conformed dates, regardless of how `Date` or `Time` are formatted.
- My own calendar displays the above linked events correctly, in spite of the disparate formatting.

I see no evidence making this change will fix the problem you describe. In fact, it may confuse people because they will have to format the date in 2 different ways.

If we're going to edit the template, I'd recommend just changing the time to 24 hour so it follows the same formatting as the UTCTime field. As far as I can tell, the top 2 fields are decorative. Did I miss something here? I didn't set any of this up. Is my analysis incorrect?

**Comment by GBKS** [comment]
@sbddesign thanks for the extra analysis. I actually did not look at the code for this update, I only did several actual tests. A other people (Mo, Bosch, and Jarol) had created call issues that did not show in the calendar (they were not included in the .ical file at all). I looked at the dates and compared them to call issues that did show up in the calendar, and noticed the difference. I tested whether adjusting the month/date order fixed the problem, and it did. 

IIRC, the top two fields were meant to be a human-readable version of the event date and time. But I think we can safely remove them since we always include an everytimezone.com link. 

Now that I think about it, it may have been an issue with the call label being added separately. The calendar script only runs when an issue is opened or edited. Assigning a label might not trigger it then....?

**Comment by sbddesign** [comment]
> Now that I think about it, it may have been an issue with the call label being added separately. The calendar script only runs when an issue is opened or edited. Assigning a label might not trigger it then....?

I think that's likely to be the case. When I create an issue, I label it while creating it. Others may label after the fact. Then those events wouldn't make it into the calendar until somebody else edits or creates an issue, thus triggering the workflow.

>  I tested whether adjusting the month/date order fixed the problem, and it did.

And so this falls in line with the labeling thing. It's not the fact that you changed the date format, it's the fact that you edited the issue in some form that caused the workflow to re-trigger. If you had opened the issue and put in a cat picture instead of changing the date, probably would have gotten same result.

So here's what I suggest:
- To fix the problem of events not appearing on calendars:
  - Update the [calendar issue workflow](https://github.com/BitcoinDesign/Meta/blob/master/.github/workflows/main.yml) to included `- labeled` on line 7.
    - No guarantees, but more likely to fix the problem.
- To improve general usability:
  - Remove `Date`, `Time`, and `Timezone` fields.
    - These are decorative. IMHO, they don't provide value, unless you happen to live in the timezone that the call is in.
    - They confuse people because people sometimes edit one and forget to edit the other.
  - Encourage usage of `UTCTime` and the EveryTimeZone link.

**Comment by GBKS** [comment]
I am not sure if everyone has rights to label issues... the other tweaks sound good.

**Comment by Bosch-0** [comment]
Encourage usage of UTCTime and the EveryTimeZone link.

Yeah this is all that is needed IMO, everything else

**Comment by GBKS** [comment]
The downside of automatically applying labels to issues is that people can use them to spam the calendar. If only admins can label stuff, there is a human review element.

**Comment by sbddesign** [comment]
> The downside of automatically applying labels to issues is that people can use them to spam the calendar. If only admins can label stuff, there is a human review element.

I'm not sure what you mean by automatically labeling issues. That's not what I am proposing at all. Let me try to explain in different words.

The core problem right now is that the workflow is only triggered when an issue is created or edited. This means that when a person with write access to the repo goes in and applies a label to the issue, the workflow will not get triggered until the next time somebody creates or edits an issue.

What I am proposing is that you add `labeled` to this list of workflow triggers. You can see a complete list of [issue related workflow triggers here](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#issues).

So when you add `labeled` to [line 7](https://github.com/BitcoinDesign/Meta/blob/fc17968bcc035882b258d430585c9aa0220317a1/.github/workflows/main.yml#L7), we'll end up with the following situation:

- Somebody who lacks write access to the repo opens an issue for a call
- Somebody who does have write access (e.g. @GBKS) sees the issue, and applies the Call label
- Calendar workflow is immediately triggered upon applying the label

Is that in-line with your security expectations?

**Comment by GBKS** [comment]
I added the `labeled` trigger, that makes a ton of sense.

As is, this PR removes the `Date`, `Time`, and `Timezone` fields, leaving only `UTCTime` and `Duration`, as those are actually used in the workflow code.

[That code](https://github.com/GBKS/github-events-calendar/blob/master/src/index.js#L51) is actually a bit messy and should be cleaned up. It references some of the fields to create an `Event` object that is not even used.

I'd appreciate another review. Thanks in advance.

**Review by sbddesign — APPROVED** [review]
Now that's the kind of edits to `design_call.md` and `main.yml` that I can get behind.

**Pull request merged**